test/testdata/desugar/forward_args.rb

```diff
@@ -18,6 +18,27 @@ def foo3(...)
   def foo4(*args, **kwargs, &block)
     T.unsafe(self).bar(*[1, 2, 3], *args, **kwargs, &block)
   end
+
+  def foo5(*)
+    [1,2,3].each do |*|
+                   # ^ error: Anonymous rest parameter
+      T.unsafe(self).p(*)
+    end
+  end
+
+  def foo6(*args)
+    [1,2,3].each do |*|
+                   # ^ error: Anonymous rest parameter
+      T.unsafe(self).p(*)
+    end
+  end
+
+  def foo7(*)
+    [1,2,3].each do |*args|
+      # This use is fine, as it refers to the anonymous rest args of the method.
+      T.unsafe(self).p(*)
+    end
+  end
```

Comment:
Running this you get a `NoMethodError` for trying to call a private method (`p`). If we make it public, there's no error here.

```ruby
require "sorbet-runtime"

Kernel.send(:public, :p)

def foo7(*)
  [1,2,3].each do |*args|
    # This use is fine, as it refers to the anonymous rest args of the method.
    T.unsafe(self).p(*)
  end
end

foo7(4, 5, 6)
```